python_orocos_kdl/PyKDL/pybind11/kinfam.cpp

```diff
@@ -175,8 +175,12 @@ void init_kinfam(pybind11::module &m)
     chain.def("addChain", &Chain::addChain);
     chain.def("getNrOfJoints", &Chain::getNrOfJoints);
     chain.def("getNrOfSegments", &Chain::getNrOfSegments);
-    chain.def("getSegment", (Segment& (Chain::*)(unsigned int)) &Chain::getSegment);
-    chain.def("getSegment", (const Segment& (Chain::*)(unsigned int) const) &Chain::getSegment);
+    chain.def("getSegment", [](const Chain &c, unsigned int nr)
```

Comment:
I am not so familiar with PyBind11, but this lambda function returns a `const Segment &` unconditionally, because the input is a `const Chain &`. Does it make a difference? Is the returned `Segment` supposed to have reference semantics or does const-correctness matter at all for the Python bindings?

The previous version had a const and a non-const overload, and the SIP version below only has a non-const implementation of `getSegment(nr)` and the const declaration has been removed.

From [this document in the `pybind11` repo](https://github.com/pybind/pybind11/blob/064a03a49be23d5b19109ca4910bf6ebdd272354/docs/limitations.rst) I understand that `const` on returned values is even cast away by `pybind11` and the return `Segment` instance is still mutable in Python, with reference semantics? In that case, would
```suggestion
    chain.def("getSegment", [](Chain &c, unsigned int nr)
```
have worked here, too, to emphasize that the returned `Segment` is mutable and the `Chain` instance can be modified through this method?